src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1266,9 +1265,10 @@
    as the factsets will have changed while the transaction was in-flight."
   [{:keys [certname producer_timestamp] :as fact-data} :- facts-schema]
   (time! (:replace-facts performance-metrics)
-         (if-let [factset-producer-ts (factset-producer-timestamp certname)]
-           (when (.before factset-producer-ts (to-timestamp producer_timestamp))
-             (update-facts! fact-data))
+         (if-let [local-factset-producer-ts (factset-producer-timestamp certname)]
```

Comment:
Is there any reason to use the factset-producer-timestamp function instead of timestamp-of-newest-record? They seem to do the same thing